### PR TITLE
DOC: Modify pandas.ExcelWriter default engine in docstring (#43359)

### LINKED ISSUE #43359: DOC: Default engine incorrect for ExcelWriter and xlsx
### 

- [X] I have checked that the issue still exists on the latest versions of the docs on `master` [here](https://pandas.pydata.org/docs/dev/)

### Location of the documentation

https://pandas.pydata.org/docs/dev/reference/api/pandas.ExcelWriter.html#pandas.ExcelWriter

### Documentation problem

Unless, I'm missing something I don't think openpyxl is the default for writing xlsx files.

```python
In [1014]: pd.ExcelWriter("foo.xlsx").engine
Out[1014]: 'xlsxwriter'

In [1015]: pd.ExcelWriter("foo.xlsx", engine="openpyxl").engine
Out[1015]: 'openpyxl'
``` 

### Suggested fix for documentation

Change the default to xlsxwriter in the docstring. Happy to submit a PR, if I'm correct.

### PR DESCRIPTION
**Pull request by LiquidMika**
* engine for xlsx : xlsxwriter instead of openpyxl
* correction to pass the doctest

- [x] closes #43359
- [x] tests added / passed : script/validate_docstrings.py detected 3 errors before my pull request. It's now resolved
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them : black pandas didn't change anything
- [x] whatsnew entry : one word changed in docstring

Script validate_docstings report : [validate_docstrings.txt](https://github.com/pandas-dev/pandas/files/7117843/validate_docstrings.txt)

### CODE REVIEW

**Review by phofl — APPROVED** [review]
lgtm

File: pandas/io/excel/_base.py [hunk 1]
```diff
@@ -647,7 +647,7 @@ class ExcelWriter(metaclass=abc.ABCMeta):
     """
     Class for writing DataFrame objects into excel sheets.
 
-    Default is to use xlwt for xls, openpyxl for xlsx, odf for ods.
+    Default is to use xlwt for xls, xlsxwriter for xlsx, odf for ods.
```
Comment (mroeschke) [inline pandas/io/excel/_base.py:650]: For `xlsx`, appears that `xlsxwriter` is preferred over `openpyxl` if installed. Might want to clarify that specifically here.

```
    _default_writers = {
        "xlsx": "openpyxl",
        "xlsm": "openpyxl",
        "xlsb": "pyxlsb",
        "xls": "xlwt",
        "ods": "odf",
    }
    assert mode in ["reader", "writer"]
    if mode == "writer":
        # Prefer xlsxwriter over openpyxl if installed
        xlsxwriter = import_optional_dependency("xlsxwriter", errors="warn")
        if xlsxwriter:
            _default_writers["xlsx"] = "xlsxwriter"
        return _default_writers[ext]
```

Comment (LiquidMika) [reply pandas/io/excel/_base.py:650]: Do you want me to change the pandas.ExcelWriter doctring this way :

    Class for writing DataFrame objects into excel sheets.

    Default is to use :
    * xlwt for xls
    * xlsxwriter for xlsx (xlsxwriter is preferred over openpyxl if both installed)
    * odf for ods.
    See DataFrame.to_excel for typical usage.

Comment (mroeschke) [reply pandas/io/excel/_base.py:650]: Sure.

Maybe clearer is `xlsxwriter for xlsx if xlsxwriter is installed otherwise openpyxl`

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @LiquidMika

**Pull request merged**